cogs/hangman.py

```diff
@@ -0,0 +1,293 @@
+            "   |    |\n"
+            "   |    O\n"
+            "   |   \n"
+            "   |   \n"
+            "   ===\n"
+            "```",  # 1 wrong
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |    O\n"
+            "   |    |\n"
+            "   |   \n"
+            "   ===\n"
+            "```",  # 2 wrong
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |    O\n"
+            "   |    |\\\n"
+            "   |   \n"
+            "   ===\n"
+            "```",  # 3 wrong
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |    O\n"
+            "   |   /|\\\n"
+            "   |   \n"
+            "   ===\n"
+            "```",  # 4 wrong
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |    O\n"
+            "   |   /|\\\n"
+            "   |     \\\n"
+            "   === \n"
+            "```",  # 5 wrong
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |    O\n"
+            "   |   /|\\\n"
+            "   |   / \\\n"
+            "   === \n"
+            "```",  # 6 wrong
+        ]
+
+    def create_buttons(self):
+        self.buttons = []
+        for letter in list(string.ascii_uppercase):
+            button_callback = self.create_button_callback(letter)
+            button = discord.ui.Button(
+                style=discord.ButtonStyle.green, label=letter, custom_id=letter
+            )
+            button.callback = button_callback
+            self.buttons.append(button)
+
+    def split_buttons_into_pages(self):
+        self.pages = [
+            self.buttons[i : i + self.page_size]
+            for i in range(0, len(self.buttons), self.page_size)
+        ]
+
+    def create_button_callback(self, button_id):
+        async def button_callback(interaction):
+            for item in self.children:
+                if item.custom_id == button_id:
+                    item.disabled = True
+                    if not self.guess_is_correct(button_id):
+                        self.incorrectLetters.append(button_id)
+            if len(self.incorrectLetters) == 6 or self.randomWord == self.revealedWord:
+                for item in self.children:
+                    # Game has ended, disable all buttons
+                    item.disabled = True
+
+            await self.updateMessage(interaction, button_id)
+
+        return button_callback
+
+    def guess_is_correct(self, guess):
+        correct = guess in self.randomWord
+        if correct:
+            self.revealedWord = "".join(
+                c if c == guess else r
+                for c, r in zip(self.randomWord, self.revealedWord)
+            )
+        return correct
+
+    async def updateMessage(self, interaction, button_id):
+        # Create a new embed with the updated information
+        incorrectLettersString = (
```

Comment:
snake case

Reply:
Resolving as the project currently uses camelCase.